### PR TITLE
Fix test failures due to Exec and FileLogger encoding issues

### LINKED ISSUE #251: Cross-platform solution for default encoding
The unit test `ExecTaskUnicodeCharacterInCommand` passes on full-framework master but fails in xplat, because the default encoding isn't available on CoreCLR.  We should fix that, and there are good ideas as to how in #246.  For now I'm going to disable the test since I'm more focused on getting failures to zero than on fixing things, and there's an easy workaround for this: don't build on paths with non-ASCII characters in them.

### PR DESCRIPTION
**Pull request by cdmihai**
Fixes #251 
Will conflict with concurrent PR #246 

Exec task:
- the default encoding is UTF-8
- the windows and unix consoles are forced to use utf-8

Filelogger:
- default encoding is UTF-8 with BOM headings

@ValMenn @AndyGerlicher @rainersigwald

### CODE REVIEW

File: src/XMakeBuildEngine/UnitTests/FileLogger_Tests.cs [hunk 1]
```diff
@@ -375,7 +381,7 @@ private void VerifyFileContent(string file, string expectedContent)
             }
         }
 
-        #region DistributedLogger
+#region DistributedLogger
```
Comment (rainersigwald) [inline src/XMakeBuildEngine/UnitTests/FileLogger_Tests.cs]: I don't know what insists on making these whitespace changes but can you revert them?  (it's been doing it to me too . . .)

File: src/XMakeTasks/Exec.cs [hunk 2]
```diff
@@ -553,21 +550,37 @@ protected internal override void AddCommandLineCommands(CommandLineBuilderExtens
 
             string batchFileForCommandLine = _batchFile;
 
-            if (NativeMethodsShared.IsWindows)
+            if (NativeMethodsShared.IsUnixLike && !NativeMethodsShared.IsMono)
```
Comment (rainersigwald) [inline src/XMakeTasks/Exec.cs]: Comment intent (especially why Mono matters here).

**Comment by rainersigwald** [comment]
LGTM once those comments are addressed.  @ValMenn are the CI build failures caused or fixed by your `ThreadPool` assembly change?

File: src/XMakeBuildEngine/UnitTests/FileLogger_Tests.cs (hunk 1 above)
Comment (cdmihai) [inline src/XMakeBuildEngine/UnitTests/FileLogger_Tests.cs]: I narrowed it down to either resharper refactorings or intellisense fixes. One of them rewrites the AST and thinks that those `#regions` should be indented.

**Comment by cdmihai** [comment]
@rainersigwald I talked with @ValMenn and indeed the isMono switch was a bug. I removed it and provided a comment for the line.

**Comment by AndyGerlicher** [comment]
LGTM

**Pull request merged**